contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	return nil
+}
+
+func (o *enableOptions) Run(cmd *cobra.Command, args []string) error {
+	// Get HiveConfig
+	hiveConfig := &hivev1.HiveConfig{}
+	if err := o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "hive"}, hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to get HiveConfig/hive")
+	}
+	if hiveConfig.Spec.AWSPrivateLink != nil {
+		log.Fatal("AWS Private Link is already enabled")
+	}
+
+	// Get active cluster's VPC, filtering by infra-id
+	describeVPCsOutput, err := o.awsClients.DescribeVpcs(&ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String(o.infraId + "-vpc")},
+			},
+		},
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get VPC of the active cluster")
+	}
+	if len(describeVPCsOutput.Vpcs) == 0 {
+		log.Fatal("VPC of the active cluster not found")
+	}
+	vpcID := *describeVPCsOutput.Vpcs[0].VpcId
+	log.Debugf("Found VPC ID = %v for the active cluster", vpcID)
+
+	// Create Hub account credentials Secret
+	hiveNS := hiveConfig.Spec.TargetNamespace
+	if hiveNS == "" {
+		hiveNS = constants.DefaultHiveNamespace
+	}
+	hubCredentialsSecret, err := o.generateAWSCredentialsSecret(hiveNS)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to generate Secret with AWS credentials")
+	}
+	if err = o.dynamicClient.Create(context.Background(), hubCredentialsSecret); err != nil {
+		log.WithError(err).Fatal("Failed to create Secret with hub account credentials")
+	}
+	log.Info("Secret with hub account credentials created")
+
+	// Update HiveConfig
+	hiveConfig.Spec.AWSPrivateLink = &hivev1.AWSPrivateLinkConfig{}
+	hiveConfig.Spec.AWSPrivateLink.AssociatedVPCs = append(
+		hiveConfig.Spec.AWSPrivateLink.AssociatedVPCs,
+		hivev1.AWSAssociatedVPC{
+			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{VPCID: vpcID, Region: o.region},
+		},
+	)
+	hiveConfig.Spec.AWSPrivateLink.CredentialsSecretRef = corev1.LocalObjectReference{Name: hubCredentialsSecret.Name}
+	hiveConfig.Spec.AWSPrivateLink.DNSRecordType = hivev1.AWSPrivateLinkDNSRecordType(o.dnsRecordType)
+	if err = o.dynamicClient.Update(context.Background(), hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to update HiveConfig")
+	}
+	log.Info("HiveConfig updated")
+
+	return nil
+}
+
+func (o *enableOptions) getRegionAndInfraId() (region string, infraId string, err error) {
+	infrastructure := &configv1.Infrastructure{}
+	if err = o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "cluster"}, infrastructure); err != nil {
+		return
+	}
+	if infrastructure.Status.PlatformStatus == nil {
+		err = errors.New("Infrastructure.status.platformStatus is empty")
+		return
+	}
+	if infrastructure.Status.PlatformStatus.AWS == nil {
+		err = errors.New("Infrastructure.status.platformStatus.aws is empty")
+		return
+	}
+
+	region = infrastructure.Status.PlatformStatus.AWS.Region
+	infraId = infrastructure.Status.InfrastructureName
+	return
```

Comment:
Curious why you're using implicit return value style for this one func.

Reply:
I should have sticked to the same style. Fixed.